### ISSUE TITLE
Error: 'stddef.h' file not found when compiling simple .cpp file which uses `#include <stdio.h>`

### ISSUE THREAD

**Issue by ackalker** [drothlis/clang-ctags#5]
(Yes, I know that C++ projects ideally should include C++ headers, but this came up in a project I'm working on which doesn't).

I'm getting an error running clang-ctags on a simple .cpp file on Arch Linux x86_64 (package clang 3.6.0-5).

```
$ clang --version
clang version 3.6.0 (tags/RELEASE_360/final)
Target: x86_64-unknown-linux-gnu
Thread model: posix
```

Steps to reproduce:
- Create `hello.cpp` with contents:

``` c++
#include <stdio.h>

int main(void) {
  printf("Hello, world!\n");
  return 0;
}
```

Compile, verify that it works:

```
$ clang -o hello hello.cpp; ./hello
Hello, world!
```

Use clang-ctags on this source file:

``` shell
$ clang-ctags -v -- clang -o hello hello.cpp
  clang parse took 0.01s
'stddef.h' file not found
clang-ctags: Error: File 'hello.cpp' failed clang's parsing and type-checking
```

This appears to be a very common issue (and even has a FAQ entry on the Clang website).
One example: http://stackoverflow.com/questions/19642590/libtooling-cant-find-stddef-h-nor-other-headers
The following answer on that SO question in particular has some more information on how to deal with it when using the Python bindings to Clang: http://stackoverflow.com/a/23314921/588561

I guess that one way to fix this would be to detect a particular Clang installation's 'built-in' include directory path at clang-ctags build-time and somehow splice it into the `compiler_command_line` string passed to `index.parse`.

**Comment by ackalker** [comment]
In my case, I can work around the issue for individual files by doing:

```shell
$ # Look up 'built-in' include path
$ clang -v -o hello hello.cpp 2>&1 | grep 'lib/clang.*include$'
 /usr/bin/../lib/clang/3.6.0/include
$ # Add it to clang-ctags command
$ $ clang-ctags -v -- clang -I /usr/bin/../lib/clang/3.6.0/include -o hello hello.cpp
  clang parse took 0.01s
  tag generation took 0.00s
```

**Comment by drothlis** [comment]
Thanks for the detailed report & investigation. I'll happily accept a pull request that adds the include path by default like the [stackoverflow answer](https://stackoverflow.com/questions/19642590/libtooling-cant-find-stddef-h-nor-other-headers/23314921#23314921) you linked.

**Comment by drothlis** [comment]
Fixed in #7.

**Comment by ackalker** [comment]
Thanks!
